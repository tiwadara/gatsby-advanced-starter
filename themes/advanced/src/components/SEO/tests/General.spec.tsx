import GeneralTags from "../General";
import { tagListHasEmptyValues, sampleSeoData } from "./TestUtils";

describe("seo module GeneralTags", () => {
  it("generates correct tags on website pages", () => {
    expect.assertions(1);

    const generatedTags = GeneralTags(
      sampleSeoData.seoWebsite,
      sampleSeoData.website
    );

    expect(generatedTags).toMatchSnapshot();
  });

  it("generates correct tags on article pages", () => {
    expect.assertions(1);

    const generatedTags = GeneralTags(
      sampleSeoData.seoArticle,
      sampleSeoData.website
    );

    expect(generatedTags).toMatchSnapshot();
  });

  it("generates correct tags when missing description", () => {
    expect.assertions(1);

    const generatedTags = GeneralTags(
      { ...sampleSeoData.seoArticle, description: undefined },
      sampleSeoData.website
    );

    expect(generatedTags).toMatchSnapshot();
  });

  it("generates correct tags when missing imageUrl", () => {
    expect.assertions(1);

    const generatedTags = GeneralTags(
      { ...sampleSeoData.seoArticle, imageUrl: undefined },
      sampleSeoData.website
    );

    expect(generatedTags).toMatchSnapshot();
  });

  it("doesn't generate empty tags", () => {
    expect.assertions(4);

    // Test for website pages
    const generatedWebsiteTags = GeneralTags(
      sampleSeoData.seoWebsite,
      sampleSeoData.website
    );

    expect(tagListHasEmptyValues(generatedWebsiteTags)).toBe(false);

    // Test for article pages
    const generatedArticleTags = GeneralTags(
      sampleSeoData.seoArticle,
      sampleSeoData.website
    );

    expect(tagListHasEmptyValues(generatedArticleTags)).toBe(false);

    // Test for missing description
    const generatedArticleTagsWithMissingDesc = GeneralTags(
      { ...sampleSeoData.seoArticle, description: undefined },
      sampleSeoData.website
    );

    expect(tagListHasEmptyValues(generatedArticleTagsWithMissingDesc)).toBe(
      false
    );

    // Test for missing imageUrl
    const generatedArticleTagsWithMissingImage = GeneralTags(
      { ...sampleSeoData.seoArticle, imageUrl: undefined },
      sampleSeoData.website
    );

    expect(tagListHasEmptyValues(generatedArticleTagsWithMissingImage)).toBe(
      false
    );
  });
});
