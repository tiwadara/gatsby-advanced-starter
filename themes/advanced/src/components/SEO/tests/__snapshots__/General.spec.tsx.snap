// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seo module GeneralTags generates correct tags on article pages 1`] = `
Array [
  <title>
    Post title
  </title>,
  <link
    href="/logos/logo-1024.png"
    rel="shortcut icon"
  />,
  <meta
    content="Post description."
    name="description"
  />,
  <meta
    content="/logos/logo-1024.png"
    name="image"
  />,
]
`;

exports[`seo module GeneralTags generates correct tags on website pages 1`] = `
Array [
  <title>
    Website title
  </title>,
  <link
    href="/logos/logo-1024.png"
    rel="shortcut icon"
  />,
  <meta
    content="Website description."
    name="description"
  />,
  <meta
    content="/logos/logo-1024.png"
    name="image"
  />,
]
`;

exports[`seo module GeneralTags generates correct tags when missing description 1`] = `
Array [
  <title>
    Post title
  </title>,
  <link
    href="/logos/logo-1024.png"
    rel="shortcut icon"
  />,
  <meta
    content="/logos/logo-1024.png"
    name="image"
  />,
]
`;

exports[`seo module GeneralTags generates correct tags when missing imageUrl 1`] = `
Array [
  <title>
    Post title
  </title>,
  <link
    href="/logos/logo-1024.png"
    rel="shortcut icon"
  />,
  <meta
    content="Post description."
    name="description"
  />,
]
`;
